cpp/src/aztec/honk/circuit_constructors/circuit_constructor_base.hpp

```diff
@@ -299,19 +305,16 @@ template <size_t program_width_> class CircuitConstructorBase {
     // uint32::MAX number of variables
     void assert_valid_variables(const std::vector<uint32_t>& variable_indices)
     {
-        for (size_t i = 0; i < variable_indices.size(); i++) {
-            ASSERT(is_valid_variable(variable_indices[i]));
+        for (const auto& variable_index : variable_indices) {
+            ASSERT(is_valid_variable(variable_index));
         }
     }
-    bool is_valid_variable(uint32_t variable_index)
-    {
-        return static_cast<uint32_t>(variables.size()) > variable_index;
-    };
+    bool is_valid_variable(uint32_t variable_index) { return variable_index < variables.size(); };
```

Comment:
The `static_cast` was not necessary since the comparison operator will promote `variable_index` to the same width as `variables.size()`.